packages/core/src/components/tree/tree.tsx

```diff
@@ -80,15 +92,16 @@ export class Tree<T = {}> extends React.Component<ITreeProps<T>, {}> {
         return this.nodeRefs[nodeId];
     }
 
-    private renderNodes(treeNodes: ITreeNode[], currentPath?: number[], className?: string): JSX.Element {
+    private renderNodes(treeNodes: Array<ITreeNode<T>>, currentPath?: number[], className?: string): JSX.Element {
         if (treeNodes == null) {
             return null;
         }
 
         const nodeItems = treeNodes.map((node, i) => {
             const elementPath = currentPath.concat(i);
+            const TypedTreeNode = TreeNode.ofType<T>();
```

Comment:
shouldn't _have_ to do this. can you revert this safely?

Reply:
This change follows the renderNodes argument type change. I believe some sort of Typed Tree Node is necessary, but I'm not sure if there is a cleaner way.